tfregistry/provider_test.go

```diff
@@ -0,0 +1,105 @@
+package tfregistry
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestProviderLatest(t *testing.T) {
+	cases := []struct {
+		desc string
+		req  *ProviderLatestRequest
+		ok   bool
+		code int
+		res  string
+		want *ProviderLatestResponse
+	}{
+		{
+			desc: "simple",
+			req: &ProviderLatestRequest{
+				Namespace: "hashicorp",
+				Type:      "aws",
+			},
+			ok:   true,
+			code: 200,
+			res:  `{"version": "3.7.0", "versions": ["3.6.0", "3.5.0", "3.7.0"]}`,
```

Comment:
Why is the slice of `versions` unsorted? I think we implicitly expect the sorted list because the code comment in the TFRegistryProviderRelease.List() function says that versions are already in asc order unlike github.

Reply:
No reason for the unsorted list, fixed.